plugins/bkpr/test/run-recorder.c

```diff
@@ -1,1249 +0,0 @@
-	db_begin_transaction(db);
-	log_chain_event(bkpr, acct,
-		make_chain_event(ctx, "to_wallet",
-				 AMOUNT_MSAT(0),
-				 AMOUNT_MSAT(200),
-				 AMOUNT_MSAT(200),
-				 blockheight + 1,
-				 '1', 1, '2'));
-
-	log_chain_event(bkpr, wal_acct,
-		make_chain_event(ctx, "deposit",
-				 AMOUNT_MSAT(150),
-				 AMOUNT_MSAT(0),
-				 AMOUNT_MSAT(150),
-				 blockheight + 1,
-				 '2', 0, '*'));
-	memset(&txid, '2', sizeof(struct bitcoin_txid));
-	maybe_update_onchain_fees(ctx, NULL, bkpr, &txid);
-	CHECK(acct->onchain_resolved_block == 0);
-
-	assert(account_onchain_closeheight(bkpr, acct) == 0);
-	CHECK(acct->onchain_resolved_block == 0);
-	db_commit_transaction(db);
-
-	/* Expect: 1 onchain fee records, all for chan-1 */
-	ofs = list_chain_fees(ctx, bkpr);
-	ofs1 = account_get_chain_fees(tmpctx, bkpr, acct->name);
-
-	CHECK(tal_count(ofs) == tal_count(ofs1));
-	CHECK(tal_count(ofs) == 1);
-
-	/* txid 4444 */
-	db_begin_transaction(db);
-	log_chain_event(bkpr, wal_acct,
-		make_chain_event(ctx, "deposit",
-				 AMOUNT_MSAT(350),
-				 AMOUNT_MSAT(0),
-				 AMOUNT_MSAT(350),
-				 blockheight + 2,
-				 '4', 0, '*'));
-
-	memset(&txid, '4', sizeof(struct bitcoin_txid));
-	maybe_update_onchain_fees(ctx, NULL, bkpr, &txid);
-
-	/* txid 3333 */
-	log_chain_event(bkpr, acct,
-		make_chain_event(ctx, "htlc_timeout",
-				 AMOUNT_MSAT(0),
-				 AMOUNT_MSAT(600),
-				 AMOUNT_MSAT(600),
-				 blockheight + 2,
-				 '1', 2, '3'));
-
-	assert(account_onchain_closeheight(bkpr, acct) == 0);
-	CHECK(acct->onchain_resolved_block == 0);
-
-	log_chain_event(bkpr, acct,
-		make_chain_event(ctx, "htlc_tx",
-				 AMOUNT_MSAT(450),
-				 AMOUNT_MSAT(0),
-				 AMOUNT_MSAT(450),
-				 blockheight + 2,
-				 '3', 0, '*'));
-
-	memset(&txid, '3', sizeof(struct bitcoin_txid));
-	maybe_update_onchain_fees(ctx, NULL, bkpr, &txid);
-
-	log_chain_event(bkpr, acct,
-		make_chain_event(ctx, "to_wallet",
-				 AMOUNT_MSAT(0),
-				 AMOUNT_MSAT(450),
-				 AMOUNT_MSAT(450),
-				 blockheight + 2,
-				 '3', 0, '4'));
-
-	memset(&txid, '4', sizeof(struct bitcoin_txid));
-	maybe_update_onchain_fees(ctx, NULL, bkpr, &txid);
-
-	db_commit_transaction(db);
-
-	/* Expect: onchain fee records for tx except channel close */
-	ofs = list_chain_fees(ctx, bkpr);
-	ofs1 = account_get_chain_fees(tmpctx, bkpr, acct->name);
-
-	CHECK(tal_count(ofs) == tal_count(ofs1));
-	CHECK(tal_count(ofs) == 3);
-
-	/* Now we update the channel's onchain fees */
-	CHECK(acct->onchain_resolved_block == 0);
-	db_begin_transaction(db);
-	account_update_closeheight(NULL, acct, account_onchain_closeheight(bkpr, acct));
-	CHECK(acct->onchain_resolved_block == blockheight + 2);
-	err = update_channel_onchain_fees(ctx, NULL, bkpr, acct);
-	CHECK_MSG(!err, err);
-	db_commit_transaction(db);
-	ofs = account_get_chain_fees(tmpctx, bkpr, acct->name);
-
-	/* Expect: fees as follows
-	 *
-	 * chan-1, 1111, 200 (anchor outs ignored)
-	 * chan-1, 2222,  50
-	 * chan-1, 3333, 150
-	 * chan-1, 4444, 100
-	 */
```

Comment:
these are pretty important tests to keep imo. it's not just database ops, it's internal fee allocation tests.

Reply:
I agree, but I ran out of time :(

It will have to enter the events in an internal sqlite3 db and simulate the calls to the sql plugin on that...

Reply:
OK, restored.  I'm not getting those hours of my life back tho!